### PR TITLE
feat:[IE-410]: Update external secrets for template-service

### PR DESCRIPTION
**Pull request by Vbhatia20**
## Describe your changes
External secrets changes for template-service

Tests executed
Build
Deployed
Validated external secrets are created without any issues.
Screenshot 2023-10-26 at 4 25 32 PM
Screenshot 2023-10-26 at 4 25 21 PM

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53674)

### CODE REVIEW

**Comment by Vbhatia20** [comment]
trigger uts

**Review by harness-jc — DISMISSED** [review]

**Comment by Vbhatia20** [comment]
trigger uts

File: template-service/chart/templates/deployment.yaml [hunk 1]
```diff
@@ -87,16 +87,19 @@ spec:
         resources:
           {{- toYaml .Values.resources | nindent 12 }}
         env:
-          {{- include "harnesscommon.dbconnection.redisEnv" (dict "context" .Values.global.database.redis "userVariableName" "EVENTS_FRAMEWORK_REDIS_USERNAME" "passwordVariableName" "EVENTS_FRAMEWORK_REDIS_PASSWORD") | indent 10 }}
-          {{- include "harnesscommon.dbconnection.redisEnv" (dict "context" .Values.global.database.redis "userVariableName" "CACHE_CONFIG_REDIS_USERNAME" "passwordVariableName" "CACHE_CONFIG_REDIS_PASSWORD") | indent 10 }}
-          {{- include "harnesscommon.dbconnection.mongoEnv" . | indent 10 }}
+          {{- $localESOSecretCtxIdentifier := (include "harnesscommon.secrets.localESOSecretCtxIdentifier" (dict "ctx" $ )) }}
```
Comment (saitejavn-harness) [inline template-service/chart/templates/deployment.yaml:90]: Unused variable?

File: template-service/chart/templates/deployment.yaml [hunk 2]
```diff
@@ -87,16 +87,19 @@ spec:
...
+          {{- include "harnesscommon.dbconnectionv2.redisEnv" (dict "ctx" $ "userVariableName" "CACHE_CONFIG_REDIS_USERNAME" "passwordVariableName" "CACHE_CONFIG_REDIS_PASSWORD") | indent 10 }}
+          {{- include "harnesscommon.dbconnectionv2.redisEnv" (dict "ctx" $ "userVariableName" "LOCK_CONFIG_REDIS_USERNAME" "passwordVariableName" "LOCK_CONFIG_REDIS_PASSWORD") | indent 10 }}
+          {{- include "harnesscommon.dbconnectionv2.mongoEnv" (dict "ctx" $) | indent 10 }}
+          {{- include "harnesscommon.secrets.manageAppEnv" (dict "ctx" . "variableName" "OPA_SERVER_SECRET" "defaultKubernetesSecretName" "template-service" "defaultKubernetesSecretKey" "OPA_SERVER_SECRET" ) | indent 10 }}
           - name : TIMESCALE_PASSWORD
             valueFrom:
               secretKeyRef:
                 name: {{ .Values.timescaleSecret.password.name }}
                 key: {{ .Values.timescaleSecret.password.key }}
           - name: MONGO_URI
```
Comment (saitejavn-harness) [inline template-service/chart/templates/deployment.yaml:101]: Why do we have only TIMESCALE_PASSWORD?

File: template-service/chart/values.yaml [hunk 3]
```diff
@@ -34,6 +34,24 @@ global:
       userKey: ""
       passwordKey: ""
       extraArgs: ""
+      secrets:
+        kubernetesSecrets:
+          - secretName: ""
+            keys:
+              MONGO_USER: ""
+              MONGO_PASSWORD": ""
```
Comment (saitejavn-harness) [inline template-service/chart/values.yaml:42]: typo: Remove " at the end

File: template-service/chart/values.yaml [hunk 4]
```diff
@@ -34,6 +34,24 @@ global:
...
+        secretManagement:
+          externalSecretsOperator:
+            - secretStore:
+                name: ""
+                kind: ""
+              remoteKeys:
+                MONGO_USER:
+                  name: ""
+                  property: ""
+                MONGO_PASSWORD":
```
Comment (saitejavn-harness) [inline template-service/chart/values.yaml:52]: typo: Remove " at the end

File: template-service/chart/values.yaml [hunk 5]
```diff
@@ -50,10 +68,44 @@ global:
...
+      secretName: ""
+      userKey: ""
+      passwordKey: ""
       extraArgs: ""
+      secrets:
+        kubernetesSecrets:
+          - secretName: ""
+            keys:
+              EVENTS_FRAMEWORK_REDIS_USERNAME: ""
+              EVENTS_FRAMEWORK_REDIS_PASSWORD": ""
```
Comment (saitejavn-harness) [inline template-service/chart/values.yaml:80]: typo: Remove " at the end

File: template-service/chart/values.yaml [hunk 6]
```diff
@@ -50,10 +68,44 @@ global:
...
+        secretManagement:
+          externalSecretsOperator:
+            - secretStore:
+                name: ""
+                kind: ""
+              remoteKeys:
+                EVENTS_FRAMEWORK_REDIS_USERNAME:
+                  name: ""
+                  property: ""
+                EVENTS_FRAMEWORK_REDIS_PASSWORD":
```
Comment (saitejavn-harness) [inline template-service/chart/values.yaml:94]: typo: Remove " at the end

File: template-service/chart/values.yaml [hunk 7]
```diff
@@ -50,10 +68,44 @@ global:
...
+                  name: ""
+                  property: ""
+                LOCK_CONFIG_REDIS_PASSWORD":
+                  name: ""
+                  property: ""
+                CACHE_CONFIG_REDIS_USERNAME":
+                  name: ""
+                  property: ""
+                CACHE_CONFIG_REDIS_PASSWORD":
+                  name: ""
```
Comment (saitejavn-harness) [inline template-service/chart/values.yaml:107]: typo: Remove " at the end

File: template-service/chart/values.yaml [hunk 8]
```diff
@@ -87,6 +139,58 @@ maxUnavailable: 0
...
+        - secretStore:
+            name: ""
+            kind: ""
+          remoteKeys:
+            REDIS_USERNAME:
+              name: ""
+              property: ""
+            REDIS_PASSWORD":
+              name: ""
+              property: ""
```
Comment (saitejavn-harness) [inline template-service/chart/values.yaml:166]: Global and Local Redis Envs do not match

File: template-service/chart/values.yaml [hunk 9]
```diff
@@ -87,6 +139,58 @@ maxUnavailable: 0
...
+    secretManagement:
+      externalSecretsOperator:
+        - secretStore:
+            name: ""
+            kind: ""
+          remoteKeys:
+            MONGO_USER:
+              name: ""
+              property: ""
+            MONGO_PASSWORD":
```
Comment (saitejavn-harness) [inline template-service/chart/values.yaml:190]: typo: Remove `"` from key name

File: template-service/chart/templates/deployment.yaml (hunk 1 above)
Comment (Vbhatia20) [reply template-service/chart/templates/deployment.yaml]: removed

File: template-service/chart/templates/deployment.yaml (hunk 2 above)
Comment (Vbhatia20) [reply template-service/chart/templates/deployment.yaml:100]: It was existing, I checked in app and it is not referred anywhere

File: template-service/chart/values.yaml (hunk 8 above)
Comment (Vbhatia20) [reply template-service/chart/values.yaml:166]: updated

File: template-service/chart/values.yaml (hunk 5 above)
Comment (Vbhatia20) [reply template-service/chart/values.yaml]: updated

File: template-service/chart/values.yaml (hunk 7 above)
Comment (Vbhatia20) [reply template-service/chart/values.yaml]: updated

File: template-service/chart/values.yaml [hunk 10]
```diff
@@ -87,6 +139,58 @@ maxUnavailable: 0
...
+  secretName: ""
+  userKey: ""
+  passwordKey: ""
+  extraArgs: ""
+  secrets:
+    kubernetesSecrets:
+      - secretName: ""
+        keys:
+          REDIS_USERNAME: ""
+          REDIS_PASSWORD": ""
```
[the remark this answers is not in the document]
Comment (Vbhatia20) [reply template-service/chart/values.yaml]: updated

**Comment by Vbhatia20** [comment]
trigger uts

**Comment by Vbhatia20** [comment]
trigger messagecheck

**Comment by Vbhatia20** [comment]
trigger gitleaks

**Comment by Vbhatia20** [comment]
trigger messagecheck

**Review by saitejavn-harness — APPROVED** [review]

**Comment by Vbhatia20** [comment]
trigger gitleaks

**Review by saitejavn-harness — APPROVED** [review]

**Pull request merged**